app/test/testmerginapi.cpp

```diff
@@ -308,6 +345,23 @@ void TestMerginApi::initTestProject()
   qDebug() << "TestMerginApi::initTestProject DONE";
 }
 
+std::shared_ptr<MerginProject> TestMerginApi::prepareTestProject()
+{
+  // Create a project
+  QString projectName = TestMerginApi::TEST_PROJECT_NAME;
+  QString projectNamespace = mUsername;
+  ProjectList projects = getProjectList();
+
+  std::shared_ptr<MerginProject> project = mApi->getProject( MerginApi::getFullProjectName( projectNamespace, projectName ) );
+  if ( project->clientUpdated < project->serverUpdated && project->serverUpdated > project->lastSyncClient.toUTC() )
+  {
+    // Fake same version on client of what is on the server
+    project->clientUpdated = project->serverUpdated;
+    project->projectDir = mProjectModel->dataDir() + projectName;
+  }
```

Comment:
Not sure if I understand this part - why do we set local version to the server version - is the project supposed to already exist on the server? From the name prepareTestProject() I expected we would be creating a new one...

Reply:
The `TestMerginApi::TEST_PROJECT_NAME` project is created in `initTestCase`. In current upload test functions, we strictly depend on existence of the project. However, these functions doesnt care, what is really upload on the server, just want to do kind of "push --force" (without downloading current version). Therefore `project->clientUpdated = project->serverUpdated;`, where we pretend, the test project is in the latest version to skip step of downloading whatever has been pushed to the project before.